intro-js.md

```diff
@@ -0,0 +1,149 @@
+
+Basically, what this means is that functions can be constructed during execution, they can be passed around as variables or parameters to other functions and you can also return a function as the value of another function (what is referred to as a currying). 
+
+Confused? That's okay. If you wish to, you can read more about it [here](http://timmknight.github.io/2015/first-class-functions-javascript/)
+
+There are a lot of other reasons, but these will be introduced as the meetings move forward. 
+
+## Basic DataTypes in JavaScript
+Some of the ECMAScript standard datatypes:
+* Boolean 
+* Null
+* Undefined
+* Number
+* String
+
+# JSON
+
+JavaScript Object Notation (JSON for short) is JavaScript object syntax. It is the primary form of information interchange on the web. 
+
+Here is a basic JSON object. 
+
+```javascript
+var jsonObj = {
+    "name": "Zaid Humayun", 
+    "age": "21", 
+    "city": "West Lafayette"
+};
+```
+
+When you have the above JSON object, getting the values for specific key value pairs is simple using the '.' operator. 
+To get the name from the above object, you would simply do
+
+```javascript
+    var name = jsonObj.name //Zaid Humayun
+```
+
+or you could also use square brackets to denote what property on the object you are trying to access
+
+```javascript
+    var name = jsonObj["name"] //Zaid Humayun
+```
+
+JSON also contains two methods called JSON.parse() and JSON.stringify() to convert between a string and JSON object. 
+
+Read more about them [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON)
+
+# Booleans in JavaScript
+
+JavaScript has two basic boolean values: true and false. 
+
+Setting some value up as a boolean is fairly simple
+
+```javascript
+var a = true //a is the true boolean value
+var b = false //b is the false boolean value
+
+//Using these values
+if(a) {
+    //do something here
+}
+if(!b) {
+    //do something else here
+}
+```
+
+But there's something weird about booleans in JS too. Who could have guessed right? 
+
+JavaScript has something called *truthy* and *falsey* values. 
+
+The following are all falsey values: 
+* undefined, null
+* Boolean: *false*
+* Number: 0, NaN
+* String: ''
+
+Every other type in JavaScript evaluates to *truthy*
+
+How do you make use of this? 
+
+```javascript
+var myStr = '' //defining an empty string here
+if(myStr) {
+    //this branch will not fire
+}else {
+    //this branch will fire
+}
+
+var a = 2 //defining a as a number here
+if(a) {
+    //this branch will fire
+}else {
+    //this branch will not fire 
+}
+```
+## Important Note
+
+You might see this form of a boolean operator used in certain places, so we wanted to introduce it to you here
+
+It is possible to typecast an object (for example) to a boolean type using a double not operator (that is, !!)
+
+```javascript 
+oObject = {'key': value} //some object here
+val = !!oObject; //typecasting to boolean to set another value
```

Comment:
Yeah, I was debating that as well. It seems like a complex topic, and I'm sure most people will just ignore it and move on. But, I'd suggest leaving it in there just for reference.